app/jobs/check_catalogues_job.rb

```diff
@@ -0,0 +1,89 @@
+class CheckCataloguesJob < ActiveJob::Base
+  queue_as :default
+
+  require 'capybara'
+  require 'capybara/poltergeist'
+  require_relative '../../app/jobs/scrape_catalogues_job.rb'
+  require_relative '../../app/jobs/delete_catalogues_job.rb'
+  require_relative '../../app/jobs/capybara_quit_job.rb'
+
+  include Capybara::DSL
+  Capybara.register_driver :poltergeist do |app|
+    Capybara::Poltergeist::Driver.new(app, timeout: 60, js_errors: false)
+  end
+  Capybara.default_driver = :poltergeist
+
+  def perform
+
+    #new_catalogues stores result from calling catalogue_gather method
+    new_catalogues = catalogue_gather
+
+    #collect current_catalogue data
+    current_catalogue = Hash.new(0)
+    Catalogue.all.each{|c| current_catalogue[c.id] = c.catalogue_num}
+
+    #scrape new posts from new catalogue
+    catalogue_to_add = add_new_catalogue(current_catalogue, new_catalogues)
+    ScrapeCataloguesJob.perform_later(catalogue_to_add)

```

Comment:
Method name `add_new_catalogue` doesn't describe what it does. Consider removing that method for now, and changing line 26 to these 2 lines:

```ruby
known_catalogue_nums = Catalogue.pluck(:catalogue_num)
catalogue_to_add = new_catalogues - known_catalogue_nums
```

There maybe more refactoring to do after that, but I think this is the first step.